src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2068,6 +2068,79 @@ void ApplicationManagerImpl::ProcessServiceStatusUpdate(
       service_type, service_event, reason, app_id);
 
   rpc_service_->ManageHMICommand(notification);
+
+  if (hmi_apis::Common_ServiceEvent::REQUEST_REJECTED == service_event &&
+      StopServicesForApp(app, service_type)) {
+    state_ctrl_.SetRegularState(app,
+                                mobile_apis::PredefinedWindows::DEFAULT_WINDOW,
+                                mobile_apis::HMILevel::HMI_NONE,
+                                true);
+  }
+}
+
+bool ApplicationManagerImpl::StopServicesForApp(
```

Comment:
Thank you for these updates, but looking at the application manager methods there are a now a few ways of "stopping" a navigation service. To avoid code redundancy, can ProcessServiceStatusUpdate or StopServicesForApp make use o the EndNaviServices Method?